cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaConfigurationDiff.java

```diff
@@ -0,0 +1,210 @@
+            "log.cleaner.backoff.ms",
+            "log.cleaner.dedupe.buffer.size",
+            "log.cleaner.delete.retention.ms",
+            "log.cleaner.io.buffer.load.factor",
+            "log.cleaner.io.buffer.size",
+            "log.cleaner.io.max.bytes.per.second",
+            "log.cleaner.max.compaction.lag.ms",
+            "log.cleaner.min.cleanable.ratio",
+            "log.cleaner.min.compaction.lag.ms",
+            "log.cleaner.threads",
+            "log.cleanup.policy",
+            "log.index.interval.bytes",
+            "log.index.size.max.bytes",
+            "log.message.timestamp.difference.max.ms",
+            "log.message.timestamp.type",
+            "log.preallocate",
+            "max.connections",
+            "max.connections.per.ip",
+            "max.connections.per.ip.overrides",
+            "log.message.downconversion.enable",
+            "metric.reporters"
+            ));
+
+    public KafkaConfigurationDiff(Map<ConfigResource, Config> current, ConfigMap desired) {
+        this.current = current;
+        this.desired = desired;
+    }
+
+    private Properties configMap2Properties() {
+        Properties result = new Properties();
+        String desireConf = desired.getData().get("server.config");
+
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(desireConf);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.replace(" ", "").startsWith("#") && !line.isEmpty())   {
+                validLines.add(line.replace(" ", ""));
+            }
+        }
+
+        return validLines;
+    }
+
+    public Set<String> getDiff() {
+        Properties desiredConfig = configMap2Properties();
+        Set<String> result = new HashSet<>();
+        Map.Entry<ConfigResource, Config> currentBrokerConfiguration = current.entrySet().iterator().next();
+
+        log.info("Diff for broker id {}", currentBrokerConfiguration.getKey());
+        currentBrokerConfiguration.getValue().entries().forEach(entry -> {
+            String val = entry.value() == null ? "null" : entry.value();
+
+            if (desiredConfig.containsKey(entry.name())) {
+                if (desiredConfig.getProperty(entry.name()).equals(val)) {
+                    // entry is in both, equal
+                } else {
+                    // entry is in both, not equal -> diff
+                    if (desiredConfig.getProperty(entry.name()) != null) {
+                        // diff is in pod dependent entry
```

Comment:
This comment is confusing. It suggests that the `if` bellow is for pod dependent eentry. But that seems to be the `else`. Please fix.